src/chainparams.cpp

```diff
@@ -248,13 +268,101 @@ class CTestNetParams : public CChainParams {
     }
 };
 
+/**
+ * SigNet
+ */
+class SigNetParams : public CChainParams {
+public:
+    explicit SigNetParams(const ArgsManager& args) {
+        std::vector<uint8_t> bin;
+        vSeeds.clear();
+        uint32_t genesis_nonce = 0;
+
+        if (!args.IsArgSet("-signet_blockscript")) {
+            LogPrintf("Using default signet network\n");
+            bin = ParseHex("512103ad5e0edad18cb1f0fc0d28a3d4f1f3e445640337489abb10404f2d1e086be43051ae");
+            genesis_nonce = 621297;
+            vSeeds.push_back("178.128.221.177");
+            vSeeds.push_back("2a01:7c8:d005:390::5");
+            vSeeds.push_back("ntv3mtqw5wt63red.onion:38333");
+        } else {
+            if (args.GetArgs("-signet_blockscript").size() != 1) {
+                throw std::runtime_error(strprintf("%s: -signet_blockscript cannot be multiple values.", __func__));
+            }
+            bin = ParseHex(args.GetArgs("-signet_blockscript")[0]);
+            genesis_nonce = args.GetArg("-signet_genesisnonce", 0);
+            if (args.IsArgSet("-signet_seednode")) {
+                vSeeds = gArgs.GetArgs("-signet_seednode");
+            }
+
+            LogPrintf("SigNet with block script %s\n", gArgs.GetArgs("-signet_blockscript")[0]);
+        }
+
+        strNetworkID = "signet";
+        g_signet_blockscript = CScript(bin.begin(), bin.end());
+        consensus.signet_blocks = true;
+        consensus.nSubsidyHalvingInterval = 210000;
+        consensus.BIP34Height = 1;
+        consensus.BIP65Height = 1;
+        consensus.BIP66Height = 1;
+        consensus.CSVHeight = 0;
+        consensus.SegwitHeight = 0;
+        consensus.nPowTargetTimespan = 14 * 24 * 60 * 60; // two weeks
+        consensus.nPowTargetSpacing = 10 * 60;
+        consensus.fPowAllowMinDifficultyBlocks = false;
+        consensus.fPowNoRetargeting = false;
+        consensus.nRuleChangeActivationThreshold = 1916;
+        consensus.nMinerConfirmationWindow = 2016;
+        consensus.powLimit = uint256S("00002adc28cf53b63c82faa55d83e40ac63b5f100aa5d8df62a429192f9e8ce5");
+        consensus.vDeployments[Consensus::DEPLOYMENT_TESTDUMMY].bit = 28;
+        consensus.vDeployments[Consensus::DEPLOYMENT_TESTDUMMY].nStartTime = 1539478800;
+        consensus.vDeployments[Consensus::DEPLOYMENT_TESTDUMMY].nTimeout = Consensus::BIP9Deployment::NO_TIMEOUT;
+
+        pchMessageStart[0] = 0xf0;
+        pchMessageStart[1] = 0xc7;
+        pchMessageStart[2] = 0x70;
+        pchMessageStart[3] = 0x6a;
+        nDefaultPort = 38333;
+        nPruneAfterHeight = 1000;
+
+        genesis = CreateSignetGenesisBlock(g_signet_blockscript, genesis_nonce);
+        consensus.hashGenesisBlock = genesis.GetHash();
```

Comment:
why not add the following here?
```
        consensus.hashGenesisBlock = genesis.GetHash();
        checkpointData = {
            {
                {0, uint256S(consensus.hashGenesisBlock.GetHex())},
            }
        };

```

Reply:
Does it change anything? (Actual question; I assumed it didn't, except for blocks > height 0)

Reply:
Well, I was thinking more on consistency with regtest, but functionally, I'm actually not sure it does anything (here or in regtest) either.